frontend/src/components/workspaces/Templates/TemplatesTableRow/TemplatesTableRow.tsx

```diff
@@ -32,31 +35,45 @@ const TemplatesTableRow: FC<ITemplatesTableRowProps> = ({ ...props }) => {
     role,
     resources,
     activeInstances,
+    createInstance,
     editTemplate,
     deleteTemplate,
+    setExpandedId,
+    expandedId,
   } = props;
 
   return (
     <>
       <div className="w-full flex justify-between py-0">
-        <Space size={'middle'}>
-          {gui ? (
-            <DesktopOutlined style={{ fontSize: '24px', color: '#1c7afd' }} />
-          ) : (
-            <CodeOutlined style={{ fontSize: '24px', color: '#1c7afd' }} />
-          )}
-          <div className="flex items-end">
-            {name}
-            {persistent && (
-              <Tooltip title="persistent">
-                <SafetyCertificateOutlined
-                  className="text-green-500 ml-2 mb-0.5"
-                  style={{ fontSize: '18px' }}
-                />
-              </Tooltip>
+        <div
+          className="flex w-full items-center cursor-pointer"
+          onClick={() =>
+            activeInstances
+              ? expandedId[0] === id
+                ? setExpandedId([''])
+                : setExpandedId([id])
+              : null
+          }
```

Comment:
I'd extract this into a function in order to improve readability.

Reply:
I did it